ex1/dates_times.py

```diff
@@ -32,13 +32,39 @@
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
-
+import calendar, datetime
+def last_of_month(dt):
+	if dt.month == 1 or dt.month == 3 or dt.month == 5 or dt.month == 7 or dt.month == 8 or dt.month == 10 or dt.month == 12:
+		return datetime.date(dt.year, dt.month, 31)
+	elif dt.month == 2 and calendar.isleap(dt.year) == True:
+		return datetime.date(dt.year, dt.month, 29)
+	elif dt. month == 2 and calendar.isleap(dt.year) == False:
+		return datetime.date(dt.year, dt.month, 28)
+	elif dt.month == 4 or dt.month == 6 or dt.month == 9 or dt.month == 11:
+		return datetime.date(dt.year, dt.month, 30)
 # Define a function named feed_the_gremlin which takes a time object as an
 # argument. It should return False between midnight and 6:30AM and True
 # otherwise.
+def feed_the_gremlin(tm):
+	if tm.hour >= 6 and tm.minute >= 30:
```

Comment:
You do not need a `if` `else` block in this case. You can just return the value of the comparison.